core/src/test/java/nl/nn/adapterframework/jdbc/dbms/ConcurrentJdbcActionTester.java

```diff
@@ -20,35 +21,34 @@ public ConcurrentJdbcActionTester(ThrowingSupplier<Connection,SQLException> conn
 
 	@Override
 	public final void initAction() throws SQLException, DbmsException {
-		connection = connectionSupplier.get();
-		initAction(connection);
+		initAction(databaseTestEnvironment);
 	}
 
-	public void initAction(Connection conn) throws SQLException, DbmsException {
+	public void initAction(DatabaseTestEnvironment databaseTestEnvironment) throws SQLException, DbmsException {
 	}
 
 
 	@Override
 	public final void action() throws SQLException {
-		action(connection);
+		action(databaseTestEnvironment);
 	}
 
-	public void action(Connection conn) throws SQLException {
+	public void action(DatabaseTestEnvironment databaseTestEnvironment) throws SQLException {
 	}
 
 
 	@Override
 	public final void finalizeAction() throws SQLException {
 		try {
-			finalizeAction(connection);
+			finalizeAction(databaseTestEnvironment);
 		} finally {
-			if (connection!=null) {
-				connection.close();
+			if (databaseTestEnvironment.getConnection() != null) {
+				databaseTestEnvironment.getConnection().close();
 			}
```

Comment:
Dit is niet goed. In deze test wordt 1 connectie herbruikt. Naast de 'setup' connection worden andere gebruikt om acties te testen. Je hoeft hier geen supplier meer mee te geven, maar je moet wel 1 connectie blijven gebruiken.

Reply:
@nielsm5 , ik begrijp je reactie niet helemaal. Dit is een Tester-class. De code hiervoor sloot al de verbinding. Dat hoeft dus niet meer, bedoel je?